core/arch/arm/mm/core_mmu.c

```diff
@@ -694,18 +694,19 @@ static void dump_mmap_table(struct tee_mmap_region *memory_map)
 }
 
 static void add_pager_vaspace(struct tee_mmap_region *mmap __maybe_unused,
-			      size_t num_elems __maybe_unused,
-			      vaddr_t begin __maybe_unused,
+			      size_t num_elems __maybe_unused, vaddr_t begin,
 			      vaddr_t end __maybe_unused,
-			      size_t *last __maybe_unused)
+			      size_t *last __maybe_unused,
+			      vaddr_t *next_pgdir_va)
 {
+	vaddr_t begin_pgdir = ROUNDDOWN(begin, CORE_MMU_PGDIR_SIZE);
 #ifdef CFG_WITH_PAGER
-	size_t size = CFG_TEE_RAM_VA_SIZE - (end - begin);
+	size_t size = CFG_TEE_RAM_VA_SIZE - (end - begin_pgdir);
```

Comment:
This is not ok. `size` is expected to cover from `begin`, not `begin_pgdir`, and `next_pgdir_va` should be `end + size` eventually rounded up to the pgdir size. With such changes, we would end to the original code (the one before this [review] patch). I even think the final roundup to pgdir size is useless.